weasyl/controllers/settings.py

```diff
@@ -52,34 +54,44 @@ def control_editprofile_get_(request):
 @login_required
 @token_checked
 def control_editprofile_put_(request):
-    form = request.web_input(
-        full_name="", catchphrase="",
-        profile_text="", set_commish="", set_trade="", set_request="",
-        set_stream="", stream_url="", stream_text="", show_age="",
-        gender="", country="", profile_display="", site_names=[], site_values=[])
+    site_names = request.params.getall('site_names')
+    site_values = request.params.getall('site_values')
+    ftrade = request.params.get('set_trade', '')
+    frequest = request.params.get('set_request', '')
+    fcommish = request.params.get('set_commish', '')
 
-    if len(form.site_names) != len(form.site_values):
+    if len(site_names) != len(site_values):
         raise WeasylError('Unexpected')
 
-    if 'more' in form:
+    sites = zip(site_names, site_values)
+
+    if 'more' in request.params:
+        form = request.params.mixed()
```

Comment:
`mixed` adds a lot of potential type confusion. I think it’s best to create a dict explicitly instead.